chia/util/keychain.py

```diff
@@ -204,11 +204,15 @@ class Keychain:
     """
 
```

Comment:
i rather we hack the tests instead, it bad practice to add production code only for testing purposes 
is it complicated to do a TestKeyringWrapper object just enhances KeyringWrapper and can be used for tests ?
you can look at what i did with BlockchainInterface to avoid using the production blockchain.py in tests

Reply:
I've updated things such that Keychain will use the shared KeyringWrapper by default, and the test code now directly overwrites the keyring_wrapper property on the keychain instance. I looked at subclassing, though I would have favored composition in this particular case. In any event, the constructor no longer takes a KeyringWrapper param.

Reply:
what i meant was that we shouldn't have test referencing in keychain.py like the testing bool param, we can open a task to change it in the future if its a lot of work

Reply:
I will address the remaining test code in a subsequent PR.